lib/src/serializable.dart

```diff
@@ -0,0 +1,224 @@
+      final registration = new _ModuleRegistration(module);
+      _registerForLifecycleEvents(registration);
+      _moduleRegistrations[module.serializableKey] = registration;
+    } else {
+      _logger.warning('Unable to serialize module without serializableKey');
+    }
+  }
+
+  void _handleApiCall(Map apiCall) {
+    String module = apiCall['module'];
+    String method = apiCall['method'];
+    Object data = apiCall['data'];
+
+    SerializableModule targetModule = _moduleRegistrations[module].module;
+
+    if (targetModule != null && data is List) {
+      _deserializeAndCall(targetModule, method, data);
+    }
+  }
+
+  void _deserializeAndCall(
+      SerializableModule module, String method, List data) {
+    InstanceMirror apiMirror = reflect(module.api);
+
+    if (apiMirror == null) {
+      _logger.warning(
+          'Unable to create mirror on api for ${module.serializableKey}');
+      return;
+    }
+
+    ClassMirror classMirror = apiMirror.type;
+    MethodMirror apiMethodMirror = classMirror.declarations[new Symbol(method)];
+
+    if (apiMethodMirror == null) {
+      _logger.warning(
+          'Method $method does not exist on ${module.serializableKey}\' module\'s API');
+      return;
+    }
+
+    // Check here that the position args in data match the expected params of the method being called
+    if (apiMethodMirror.parameters.length == data.length) {
+      for (var i = 0; i < apiMethodMirror.parameters.length; i++) {
+        var param = apiMethodMirror.parameters[i];
+
+        if (data[i] is Map && param.type.reflectedType != Map) {
+          ClassMirror paramClassMirror = reflectClass(param.type.reflectedType);
+
+          // Paramter type must implement fromJson name constructor that takes a Map
+          try {
+            var instance = paramClassMirror
+                .newInstance(new Symbol('fromJson'), [data[i]]).reflectee;
+            data[i] = instance;
+          } on NoSuchMethodError {
+            _logger.warning(
+                '${paramClassMirror.simpleName.toString()} does not implement fromJson named constructor');
+            return;
+          }
+        }
+      }
+
+      try {
+        apiMirror.invoke(new Symbol(method), data);
+      } catch (e) {
+        _logger.severe(
+            'Unable to call $method on ${module.serializableKey} w_module, ${e.toString()}');
+      }
+    } else {
+      _logger.warning(
+          'Unable to call api method $method in ${module.serializableKey} w_module, mismatched params');
+    }
+  }
+
+  void _registerForAllEvents(_ModuleRegistration registration) {
+    if (registration.module.events != null) {
+      for (var event in registration.module.events.allEvents) {
+        if (event is SerializableEvent) {
+          registration.manageStreamSubscription(event.listen((payload) =>
+              _sendEvent(registration.module, event.eventKey, payload)));
+        }
+      }
+    } else {
+      _logger.info(
```

Comment:
In an earlier comment @evanweible-wf suggested that `info` were to be used because what it is logging is not a required for a module to implement.